File: app/src/main/java/com/example/techrepairs/TechListFragment.java
package com.example.techrepairs;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;


public class TechListFragment extends Fragment {
    String[] tech = { "Nokia 3310", "Xiaomi MI A3", "Iphone X", "Huawei P 30 Pro", "Nokia 3330"};

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_tech_list, container, false);
        ListView contactsList = (ListView) view.findViewById(R.id.techList);
        final Intent intent = new Intent(view.getContext(), TechDetails.class);

        ArrayAdapter<String> adapter = new ArrayAdapter(view.getContext(),
                android.R.layout.simple_list_item_1, tech);

        contactsList.setAdapter(adapter);

        contactsList.setOnItemClickListener(new AdapterView.OnItemClickListener(){
            @Override
            public void onItemClick(AdapterView<?> parent, View v, int position, long id)
            {
                startActivity(intent);
            }
        });

        return  view;
    }
}

File: app/src/main/java/com/example/techrepairs/techlist/TechListFragment.kt
package com.example.techrepairs.techlist;

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.example.techrepairs.R
import com.example.techrepairs.techdetails.TechDetails


class TechListFragment : Fragment() {

    private lateinit var viewModel: TechListViewModel
    private  lateinit var viewModelFactory: TechListViewModelFactory

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        var view : View = inflater?.inflate(R.layout.fragment_tech_list, container, false);

        viewModelFactory = TechListViewModelFactory()
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(TechListViewModel::class.java)


        var techList:ListView =  view?.findViewById(R.id.techList);

        var adapter = ArrayAdapter<String>(view.getContext(),
                android.R.layout.simple_list_item_1, viewModel.tech);

        techList.setAdapter(adapter);
        techList.setOnItemClickListener { parent, view, position, id ->
            activity?.supportFragmentManager?.beginTransaction()?.replace(R.id.activity_main, TechDetails())?.commit()
         }

        return  view;
    }
}